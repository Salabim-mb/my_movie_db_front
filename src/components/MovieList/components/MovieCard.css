.MovieCard {
    text-align: justify;
    margin-left: 3%;
}

.ThumbnailImage {
    max-width: 180px;
}