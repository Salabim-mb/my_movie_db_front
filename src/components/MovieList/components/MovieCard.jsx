import React from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import './MovieCard.css';

const MovieCard = ({movie}) => (
    <div className="card mb-3">
        <div className="row no-gutters">
            <div className="ThumbnailImage">
                <Card.Img src={movie.movieImage}/> {/*{movie.movieImage}*/}
            </div>
            <div className="col-sm-9">
                <div className="MovieCard">
                    <div className="card-body">
                        <Card.Title>
                            {movie.movieName}
                        </Card.Title>
                        <Card.Subtitle className="mb-2 text-muted">
                            {movie.movieDirector}
                        </Card.Subtitle>
                        <p className="card-text"><small className="text-muted">{movie.movieGenre}</small></p>
                        <Card.Text>
                            {movie.movieReleaseYear}
                        </Card.Text>
                        <hr/>
                        <p className="card-text">{movie.movieDescription} </p>
                    </div>
                </div>
            </div>
        </div>
    </div>

);

export default MovieCard;