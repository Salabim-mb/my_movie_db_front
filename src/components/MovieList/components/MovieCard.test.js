import React from "react";
import MovieCard from "./components/MovieCard";
import {render, waitForElement} from '@testing-library/react';



describe("MovieCard", () => {
    let apiFail;
    let res;

    beforeAll(() => {
        global.fetch = jest.fn().mockImplementation((input, init) => {
            return new Promise((resolve) => {
                if (apiFail) {
                    resolve({status: 500});
                } else {
                    resolve({status: 200, json: () => Promise.resolve(res)});
                }
            });
        });
    });

    beforeEach(() => {
        jest.clearAllMocks();
        MovieCard.mockImplementation((props) => <div>{props?.movie?.title}</div>);
        apiFail = false;
        res = [
            {
                id: 1,
                title: "Titanic"
            }, {
                id: 2,
                title: "Batman"
            }
        ]
    });

    it("should match snapshot", async () => {
        const {container, getByText} = render(
            <MemoryRouter>
                <MovieList />
            </MemoryRouter>
        );

        await waitForElement(() => getByText("Titanic"));

        expect(container).toMatchSnapshot();
    });

    it("should fetch correct data", async () => {
        const {getByText, queryByText} = render(
            <MemoryRouter>
                <MovieList />
            </MemoryRouter>
        );
        await waitForElement(() => getByText("Titanic"));

        expect(getByText("Titanic")).toBeInTheDocument();
        expect(queryByText("Something went wrong", {exact: false})).toBe(null);
        //expect(queryByText("Something went wrong", {exact: false})).not.toBeInTheDocument();
    });

    it ("should render error alert", async () => {
        apiFail = true;

        const {getByText, queryByText} = render(
            <MemoryRouter>
                <MovieList />
            </MemoryRouter>
        );

        await waitForElement(() => getByText("Something went wrong", {exact: false}));

        expect(getByText("Something went wrong", {exact: false})).toBeInTheDocument();
        expect(queryByText("Batman")).not.toBeInTheDocument();
    });
});