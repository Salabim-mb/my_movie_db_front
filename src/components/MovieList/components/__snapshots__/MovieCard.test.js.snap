// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match snapshot 1`] = `
<div>
  <div
    class="card mb-3"
  >
    <div
      class="row no-gutters"
    >
      <div
        class="ThumbnailImage"
      >
        <img
          alt="Movie image"
          class="card-img"
          src="someImageInBase64"
        />
      </div>
      <div
        class="Button"
      >
        <a
          class="btn btn-warning btn-sm"
          href="/movies/new/3"
        >
          Edit
        </a>
        <div
          class="ButtonRemove"
        >
          <button
            class="btn btn-danger btn-sm"
            type="button"
          >
            Remove
          </button>
        </div>
      </div>
      <div
        class="col-sm-9"
      >
        <div
          class="MovieCard"
        >
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              KOD NIEŚMIERTELNOŚCI
            </div>
            <div
              class="mb-2 text-muted card-subtitle h6"
            >
              Duncan Jones
            </div>
            <p
              class="card-text"
            >
              <small
                class="text-muted"
              >
                Thriller / Sci-Fi
              </small>
            </p>
            <p
              class="card-text"
            >
              2011
            </p>
            <hr />
            <p
              class="card-text"
            >
              Kapitan Colter Stevens budzi się, nie pamiętając niczego. Wkrótce odkrywa, że bierze udział w misji, której celem jest złapanie terrorysty podkładającego bomby na terenie Chicago.
               
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
